crates/rpc/src/handlers/validator.rs

```diff
@@ -424,3 +425,40 @@ fn check_validator_participation(
         Ok(validator.is_active_validator(epoch))
     }
 }
+#[get("/validator/attestation_data")]
+pub async fn get_attestation_data(
+    db: Data<ReamDB>,
+    query: Query<AttestationQuery>,
+) -> Result<impl Responder, ApiError> {
+    let beacon_block_root = db
+        .slot_index_provider()
+        .get_highest_root()
+        .map_err(|err| ApiError::InternalError(format!("Failed to slot_index, error: {err:?}")))?
+        .ok_or(ApiError::NotFound(
+            "Failed to find highest block root".to_string(),
+        ))?;
+
+    let source = db
+        .justified_checkpoint_provider()
+        .get_justified_checkpoint()
+        .map_err(|err| {
+            ApiError::InternalError(format!("Failed to get source checkpoint, error: {err:?}"))
+        })?;
+
+    let target = db
+        .unrealized_justified_checkpoint_provider()
+        .get_unrealized_checkpoint()
+        .map_err(|err| {
+            ApiError::InternalError(format!("Failed to target checkpoint, error: {err:?}"))
+        })?;
+
+    let data = AttestationData {
+        slot: query.slot,
+        index: query.committee_index,
+        beacon_block_root,
+        source,
+        target,
+    };
+
+    Ok(HttpResponse::Ok().json(BeaconResponse::new(data)))
```

Comment:
```suggestion
    Ok(HttpResponse::Ok().json(DataResponse::new(data)))
```

I think we should use `DataResponse` to be consistent with the spec